parodos-model-api/src/test/java/com/redhat/parodos/infrastructure/option/InfrastructureOptionsTest.java

```diff
@@ -0,0 +1,95 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.parodos.infrastructure.option;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+public class InfrastructureOptionsTest {
+
+    private final InfrastructureOption option = new InfrastructureOption.Builder("testOption", "testWorkFlowDefinition")
+            .displayName("testWorkFlowDefinition").addToDetails("An awesome WorkFlow").build();
+
+    @Test
+    public void verifyInfraOptions() {
+        InfrastructureOptions options = new InfrastructureOptions.Builder().addNewOption(option).build();
+
+        assertTrue(options.getNewOptions().size() == 1);
+        assertEquals(option, options.getNewOptions().get(0));
+        assertFalse(options.isOptionsAvailable());
+    }
+
+    @Test
+    public void checkOptionAvailability() {
+        InfrastructureOptions options = new InfrastructureOptions.Builder().addNewOption(option)
+                .addContinuationOption(option).addMigrationOption(option).addUpgradeOption(option).build();
+
+        assertTrue(options.getContinuationOptions().size() == 1);
+        assertTrue(options.getMigrationOptions().size() == 1);
+        assertTrue(options.getNewOptions().size() == 1);
+        assertTrue(options.getUpgradeOptions().size() == 1);
+        assertTrue(options.isOptionsAvailable());
+    }
+
+    @Test
+    public void verifyContinuation() {
+        InfrastructureOptions options = new InfrastructureOptions.Builder().build();
+
+        assertTrue(options.getContinuationOptions().size() == 0);
+        assertFalse(options.hasIncompleteWorkFlow());
+
+        options.addContinuationOption(option);
+
+        assertTrue(options.getContinuationOptions().size() == 1);
+        assertTrue(options.hasIncompleteWorkFlow());
+    }
+
+    @Test
+    public void verifyMigration() {
+        InfrastructureOptions options = new InfrastructureOptions.Builder().build();
+
+        assertTrue(options.getMigrationOptions().size() == 0);
+
+        options.addMigrationOption(option);
+
+        assertTrue(options.getMigrationOptions().size() == 1);
+    }
+
+    @Test
+    public void verifyUpgrade() {
+        InfrastructureOptions options = new InfrastructureOptions.Builder().build();
+
+        assertTrue(options.getUpgradeOptions().size() == 0);
+
+        options.addUpgradeOption(option);
+
+        assertTrue(options.getUpgradeOptions().size() == 1);
+    }
+
+    @Test
+    public void checkInfra() {
+        InfrastructureOptions options = new InfrastructureOptions.Builder().setCurrentInfrastructure(option).build();
+
+        assertFalse(options.hasInfrastructure());
```

Comment:
Is this correct?

```java
	public boolean hasInfrastructure() {
		return currentVersion == null;
	}
```

Either the name is confusing or we have a bug

Reply:
The name is confusing. This should be indicating if we were able to determine if the project already had a Workflow applied to it. What I mean by this is the project might be on OCP3 already. So in the UI we would display the project is on OCP3, and then perhaps give them an option to upgrade to OCP4